python/src/viewer.py

```diff
@@ -638,77 +740,100 @@ def ToImageString(graph):
     return image_string
 
 
-def PlotDesign(design, bounds, Nx, Ny, figure=None, axes=[], plot_kwargs={}, axes_kwargs={}, text_kwargs={}):
+def PlotDesign(design, bounds = None, subdivisions = None, 
+               figure = None, axes=[], 
+               plot_kwargs = None, axes_kwargs = None, text_kwargs = None, 
+               enableTicks = True):
     """
-    Plot a design using a scatter plot approach (plots 2D marginals).
+    Plot a design using a scatter plot. 
+    If the dimension is equal to 2, then plots the 2D projection. 
+    If the dimension is greater or equal to 3, then plots all 2D projections.
 
     In addition, the function plots a grid, i.e. horizontal and vertical lines
-    to distinguish LHS character
+    to see the dispersion of the points. 
+    This allows to see how the sample fills the space. 
 
     Parameters
     ----------
     design : 2-d sequence of float
-        Design
+        The sample. 
+    figure : a Matplotlib figure. 
+        If this is not None, then create a new figure. 
+        Otherwise, use the existing figure. 
+    axes : a Matplotlib axis. 
+        If empty, then create new axes. 
     bounds: :class:`openturns.Interval`
-        Bounds of the underlying distribution
-    nxdiv : int
-        Number of subdivisions in the X axis
-    nydiv : int
-        Number of subdivisions in the Y axis
+        Bounds of the plot. By default, compute the bounds from the sample. 
+    subdivisions : a list of integers
+        Number of subdivisions in the each direction. 
+        By default, set the number of subdivisions in each direction 
+        as equal to the sample size. 
+    enableTicks :
+        A boolean. If True, then the ticks are plotted. 
 
     Returns
     -------
     fig : matplotlib figure
-          Figure representing the LHS
+          Figure representing the sample.
```

Comment:
yes and no. In theory we need a `Sample` but in practice only small `LHS` structures in `uniform` space should be considered. Otherwise there is no interest of using this function comparing to a scatter matrix (`Pairs`)

Reply:
I removed LHS because this plot can be useful for other types of designs, such as Quasi-Monte-Carlo designs for example. This is shown in the following example I just created in this PR: 

https://github.com/mbaudin47/openturns/blob/09bf303037d9c4469f9ea1e0994bf8e395e81cb4/python/doc/examples/reliability_sensitivity/design_of_experiments.ipynb

Unfortunately, the example is wrong: I must fix it.

Reply:
Please look at the Sobol' and Halton plots here: https://github.com/mbaudin47/openturns/blob/FixDocForExamples/python/doc/examples/reliability_sensitivity/design_of_experiments.ipynb This shows how the `PlotDesign` method can be useful a  types of DOE which is not a LHS.